src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4227,11 +4223,7 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             else
             {
                 TypeSymbolWithAnnotations rightType = VisitOptionalImplicitConversion(right, leftType, UseLegacyWarnings(left), AssignmentKind.Assignment);
-                int rightSlot = MakeSlot(right);
-                int slotWatermark = GetSlotWatermark();
-                // leftSlot is calculated last to avoid copying from that slot in cycles.
-                int leftSlot = MakeSlot(left);
-                TrackNullableStateForAssignment(right, leftType, leftSlot, rightType, rightSlot, slotWatermark);
+                TrackNullableStateForAssignment(right, leftType, MakeSlot(left), rightType, MakeSlot(right));
```

Comment:
>MakeSlot [](start = 92, length = 8)

Unrelated to this PR directly, but `MakeSlot` feels like a bad name for this API. We're not always making the slot, sometimes we're retrieving the existing slot.

Reply:
Yes, perhaps the method should be called `GetOrCreateSlot`. Will leave as is though since this is an override of an existing virtual method also used by definite assignment.

---
In reply to: [258163043](https://github.com/dotnet/roslyn/pull/33455#discussion_r258163043) [](ancestors = 258163043)